lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -50,10 +50,6 @@ def timeout_message
       @pods.map(&:timeout_message).compact.uniq.join("\n")
     end
 
-    def deploy_timed_out?
```

Comment:
Can you comment why we should fall back to 

https://github.com/Shopify/kubernetes-deploy/blob/09f6121b2f02f1fbb51fb582e63f077328d74d81/lib/kubernetes-deploy/kubernetes_resource.rb#L113-L116

not to `@pods.all?(&:deploy_timed_out?)`?

Reply:
The most pragmatic reason is that `@pods.all?(&:deploy_timed_out?)` will never be true before `super` anyway, since the Pod timeout is double the RS timeout; this is just making the behaviour more difficult to understand with no real benefit. In fact, the Pod hard timeout was chosen based on the needs of bare pods, which are assumed to be running a single high-value pseudo-inline task, so referencing it from RS doesn't really make sense. Finally, the supported, k8s-native way to time out a deployment whose pods aren't rolling in time is using `progressDeadlineSeconds`.